lib/galaxy/jobs/__init__.py

```diff
@@ -786,6 +790,10 @@ def get_parallelism(self):
         return self.tool.parallelism
 
     @property
+    def shell(self):
+        return self.job_destination.shell or getattr(self.app.config, 'default_job_shell', DEFAULT_JOB_SHELL)
```

Comment:
In this particular case because the property is a string and the config property matches the name in galaxy.ini - these two are likely equivalent. But for booleans, numbers, etc... the config object attributes and not the dictionary contains the processed value - it is however for this reason that I the canonical source of the information should be assumed to be the object attribute.

Most places in the code that I have seen access config options as attributes and via ``.get()`` - I don't think this is a problem and I don't think we promise ``get()`` does anything other than access the raw values in a dictionary like way. Instead of modifying ``get``, I would rather see code that accesses config options this way access them via attributes instead.